DB 89 tutorials (#119)
elasticpath/react-pwa-reference-storefront: documentation/tutorial_creatingPages.md

  d) Create a new `.jsx` file in the `src/containers/<PageName>` directory. <br>
      For example, `src/containers/MyPage/MyPage.jsx`.<br>
  e) To populate the page with the required structure, copy the contents of an existing page to the new page.<br>
  f) In the `src/components/routes.js` directory:<br>
    i)  Import the new page.<br>
    ii) Define the routing path for the page.<br>
  g) In the `.jsx` file:<br>
    * To view the changes in the storefront, update the export settings with the page name.<br>
    * Add all required components and content.<br>
  h) Add the required custom CSS in the `src/containers/<PageName>/<PageName>.less` directory. <br>
  <br>
2. Create new components in React.<br>
  a) In the `src/components` directory, create a new component directory with the appropriate name.<br>
  Name of the directory and component must be the same. For example, `mycomponent`.<br>
  b) Create a new `.jsx` file in the `src/components/<componentName>` directory. <br> For example, `src/components/mycomponent.main.jsx`.<br>

>   d) To populate the file with the required structure, copy the contents of an existing component to the new component.<br>

Author: I've updated the "Populate" part but I am a bit fuzzy on if you want me to link to the page as well?  I think you are just explaining, but please let me know if I need to add any more info :)

>   d) To populate the file with the required structure, copy the contents of an existing component to the new component.<br>

Reviewer: No, this is only a comment. We dont need a link here. the additional details were to help you understand :)

# Tutorial: Create Reference Storefront Pages and Components

### Overview

> This tutorial demonstrates how to create the pages and components for a React PWA Reference Storefront. The React PWA Reference Storefront is composed of containers (pages) and React components. Each page in the storefront is composed of various components that fulfills the purpose of the page. For example, home, cart, categories, checkout, and registration are separate pages. Each page consists of various components, such as navigation, footer, product list, or products.

Reviewer 2: And by link to the reference storefront overview, it should one that resides within the github page and doesn't take you back out to developers.elasticpath.com

### Prerequisites
This tutorial requires an Elastic Path developer environment: 

* Your own development environment
* An Elastic Path Training virtual machine

Ensure that you are familiar with the following third-party technologies:

* Git
* Node.js
* Visual Studio Code with the following extensions:

> *   * Debugger for Chrome

Reviewer 3: I love bullet points too.... but two per line are kinda unnecessary.
```suggestion
*   Debugger for Chrome
```

*   * ESLint extension
* Javascript
* React

### Business Requirements

As a developer, I want to create pages and components for a React PWA Reference Storefront experience so my company can engage customers on multiple touchpoints, such as desktop, tablet and mobile devices.

### Reference Example

The goal of this tutorial is to create a Reference Storefront container and components for a **Terms and Conditions** page.

### Exercise

#### Step 1: Create New Containers in React

To create pages for the storefront, first you have to create containers as a structure for the pages. 

1. From the command line, navigate to the `src/containers` directory. Create a new folder for the page.

> 2. Name the directory. <br>**Note**: Ensure that the name of the directory and the correpsonding page are the same. For example, if the name of the page is `MyPage`, name the containter `MyPage`.

Author: Why are we removing this? Do you feel it's redundant?

> 3. In the `src/containers/<PageName>` directory, create a new `.jsx` file for the page. For example, `src/containers/MyPage/MyPage.jsx`.

Reviewer: ```suggestion
3. In the `src/containers/<PageName>` directory, create a new `.jsx` file. </br> For example, `src/containers/MyPage/MyPage.jsx`.
```

> 4. Copy the contents of an existing page to the new page to populate the page with the required structure.

Reviewer: To populate the page with the required structure, copy the contents of an existing page to the new page.

> 5. Name the class of the page with the page name.

Reviewer: I dont get this step

> 6. In the `src/components/routes.js` directory:

Reviewer: `src/components/routes.js` it this  `src/containers/....'?

Author: This has been updated.

    * Import the new page.

>     * Define the desired routing path for the page.

Reviewer: ```suggestion
    * Define the routing path for the page.
```

> 7. In the `.jsx` file of the page:

Reviewer: ```suggestion
7. In the `.jsx` file:
```

>   * Update the export settings of the page with the page name to view the changes in the storefront.

Reviewer: ```suggestion
  * To view the changes in the storefront, update the export settings with the page name .
```

>   * Add all required components and content for the page.

Reviewer: ```suggestion
  * Add all required components and content.
```

8. In the `src/containers/<PageName>/<PageName>.less` directory, add the required custom CSS.

#### Step 2: Create New Components in React

> Create components so you have reusable pieces you can use in the storefront. For example, a component can be the navigation, footer, product list, or products.

Reviewer: ```suggestion

```

> 1. In the `src/components` directory, create a new component for the page.

Reviewer: ```suggestion
1. In the `src/components` directory, create a new component directory with appropriate name.
<br> Name of the directory and component directory must be the same. For example, `mycomponent`.
```

> 2. Name the directory. <br> Name of the directory and component must be the same. For example, `mycomponent`.

Reviewer: ```suggestion

```

> 3. In the `src/components/<componentName>` directory, create a new `.jsx` file for the page. For example, `src/components/mycomponent.main.jsx`.

Reviewer: ```suggestion
3. In the `src/components/<componentName>` directory, create a new `.jsx` file. <br/>For example, `src/components/mycomponent.main.jsx`.
```

> 4. Copy the contents of an existing component to the new component to populate the component with the required structure.

Reviewer: see comment in the above section.

5. Name the class of the component with the component name.
6. In the jsx file of the component:
* Update the export settings of the page with the component name to view the changes in the storefront.
* Add all required components and content for the component.
* Import the component to other components or pages as required.

> 7. In the `src/components/mycomponent.main.jsx` file, add the required custom CSS.

Reviewer 3: Styles should be added to the .less file

Author: @shaunmaharaj is the file name wrong then for # 7?

Reviewer 3: @asmurthwaite yes and no... I'll make the change to correct this and push to your branch.

### Example

In this section, [Extending React Reference Storefront](https://elasticpath.github.io/react-pwa-reference-storefront/documentation/extending/) is explained with an example.

1. Navigate to `src/containers` directory.<br>
2. Create a new folder.<br>
3. Name the directory. <br>

>     **Note**: Ensure that the name of the container directory and components directory are the same as the correpsonding directories. For example, if the name of the page is `MyPage`, name the containter `MyPage`.<br>

Reviewer: mmmm, my suggestion was to add it before the task starts.

  * Add all required components and content.<br>
8. Add the required custom CSS in the `src/containers/<PageName>/<PageName>.less` directory. <br>
9. In the `src/components` directory, create a new component directory with the appropriate name.<br>

>   **Note** Ensure that the name of the container directory and components directory are the same as the corresponding directories. For example, for the page `MyPage`, name the container `MyPage`.<br>

Reviewer: I suggested a generic note to avoid repeating this note
